### PR TITLE
Zip `cdt_name` with `docker_image`

### PR DESCRIPTION
**Pull request by jakirkham**
Second attempt at adding `cdt_name` following the first one in PR ( https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/867 ).

Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by jakirkham** [comment]
cc @isuruf @beckermr

File: recipe/migrations/cuda110.yaml [hunk 1]
```diff
@@ -41,6 +41,18 @@ cudnn:
...
+- cos6  # [os.environ.get("BUILD_PLATFORM") == "linux-64" or (os.environ.get("CONFIG_VERSION", "1") == "1" and linux64)]
+- cos7  # [os.environ.get("BUILD_PLATFORM") == "linux-aarch64" or (os.environ.get("CONFIG_VERSION", "1") == "1" and aarch64)]
+- cos7  # [os.environ.get("BUILD_PLATFORM") == "linux-ppc64le" or (os.environ.get("CONFIG_VERSION", "1") == "1" and ppc64le)]
+- cos7  # [os.environ.get("BUILD_PLATFORM") == "linux-armv7l" or (os.environ.get("CONFIG_VERSION", "1") == "1" and armv7l)]
+
+- cos6  # [linux64 and (os.environ.get("BUILD_PLATFORM") == "linux-64" or os.environ.get("CONFIG_VERSION", "1") == "1")]
+- cos6  # [linux64 and (os.environ.get("BUILD_PLATFORM") == "linux-64" or os.environ.get("CONFIG_VERSION", "1") == "1")]
+- cos6  # [linux64 and (os.environ.get("BUILD_PLATFORM") == "linux-64" or os.environ.get("CONFIG_VERSION", "1") == "1")]
+- cos6  # [linux64 and (os.environ.get("BUILD_PLATFORM") == "linux-64" or os.environ.get("CONFIG_VERSION", "1") == "1")]
+- cos7  # [linux64 and (os.environ.get("BUILD_PLATFORM") == "linux-64" or os.environ.get("CONFIG_VERSION", "1") == "1")]
```
Comment (isuruf) [inline recipe/migrations/cuda110.yaml:54]: Can you make the selectors identical to the ones in the main pinning file?

**Review by isuruf — CHANGES_REQUESTED** [review]

File: recipe/migrations/cuda110.yaml (hunk 1 above)
Comment (jakirkham) [reply recipe/migrations/cuda110.yaml:54]: Yeah sorry I think I cherry-picked the old commits and edited. So some things didn't get updated correctly. Will fix.

Comment (jakirkham) [reply recipe/migrations/cuda110.yaml]: Alright should be fixed now. PTAL :)

**Comment by beckermr** [comment]
We good to merge @isuruf ?

File: recipe/conda_build_config.yaml [hunk 2]
```diff
@@ -146,6 +146,7 @@ zip_keys:
   -                             # [linux64]
     - cudnn                     # [linux64]
     - cuda_compiler_version     # [linux64]
+    - cdt_name                  # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64"]
```
Comment (isuruf) [inline recipe/conda_build_config.yaml:149]:
```suggestion
    - cdt_name                  # [linux64]
```

File: recipe/migrations/cuda110.yaml [hunk 3]
```diff
@@ -57,3 +69,4 @@ zip_keys:
   - - cudnn                      # [linux64]
     - cuda_compiler_version      # [linux64]
     - docker_image               # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64"]
+    - cdt_name                   # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64"]
```
Comment (isuruf) [inline recipe/migrations/cuda110.yaml:72]:
```suggestion
    - cdt_name                   # [linux64]
```

**Comment by jakirkham** [comment]
Thanks Isuru! Should be fixed now 😄

**Comment by isuruf** [comment]
Looks good. Have you checked that this works with a non cuda feedstock and a cuda enabled feedstock?

**Comment by jakirkham** [comment]
Nope 🙂 it might be a bit before I’m able to do this so if someone has spare cycles and would like to test that would be welcome 😉

**Comment by jakirkham** [comment]
cc @jaimergp

**Comment by jaimergp** [comment]
Seems to be working!

* With CUDA (OpenMM): https://github.com/jaimergp-feedstocks/openmm-feedstock/tree/migration-test
* Without CUDA (AmberTools): https://github.com/jaimergp-feedstocks/ambertools-feedstock/tree/cuda-migration-test

**Comment by jaimergp** [comment]
Ah wait, the non-CUDA repos will have their compiler versions [downgraded](https://github.com/jaimergp-feedstocks/ambertools-feedstock/commit/6731bcd5869befe11b0daa8c8940e1bcca60fad7#diff-0f09425ff69e95e936fefb8fb68a54747b4647b2a35a77d28198d2af31ac5207R10) with this though...

**Comment by isuruf** [comment]
> Ah wait, the non-CUDA repos will have their compiler versions downgraded with this though...

@jaimergp, why is `.ci_support/migrations/cuda110.yaml` in that feedstock?

**Comment by isuruf** [comment]
For non-CUDA builds, we should check by doing `conda-smithy rerender -e <path to conda_build_config.yaml from this pr>`

**Comment by jaimergp** [comment]
Ah, my bad. I thought we were testing the situation where the migrator accidentally ended up there, but it shouldn't have any effect. I'll run that command now.

**Comment by jaimergp** [comment]
I removed `cuda110.yaml` from `.ci_support/migrations`, rerendered to have a good baseline and then ran the command, which resulted in an error:

```
% conda-smithy rerender -e ../cuda110.yaml
Setting build platform. This is only useful when pretending to be on another platform, such as for rendering necessary dependencies on a non-native platform. I trust that you know what you're doing.
WARNING:conda_build.config:Setting build platform. This is only useful when pretending to be on another platform, such as for rendering necessary dependencies on a non-native platform. I trust that you know what you're doing.
Setting build arch. This is only useful when pretending to be on another arch, such as for rendering necessary dependencies on a non-native arch. I trust that you know what you're doing.
WARNING:conda_build.config:Setting build arch. This is only useful when pretending to be on another arch, such as for rendering necessary dependencies on a non-native arch. I trust that you know what you're doing.
No numpy version specified in conda_build_config.yaml.  Falling back to default numpy value of 1.11
WARNING:conda_build.metadata:No numpy version specified in conda_build_config.yaml.  Falling back to default numpy value of 1.11
Adding in variants from internal_defaults
INFO:conda_build.variants:Adding in variants from internal_defaults
Adding in variants from /home/jaime/devel/py/jaimergp/cuda110.yaml
INFO:conda_build.variants:Adding in variants from /home/jaime/devel/py/jaimergp/cuda110.yaml
Adding in variants from /home/jaime/devel/py/jaimergp/ambertools-feedstock/recipe/conda_build_config.yaml
INFO:conda_build.variants:Adding in variants from /home/jaime/devel/py/jaimergp/ambertools-feedstock/recipe/conda_build_config.yaml
INFO:conda_smithy.configure_feedstock:Applying migrations: /home/jaime/devel/py/jaimergp/ambertools-feedstock/.ci_support/migrations/python39.yaml
Traceback (most recent call last):
  File "/home/jaime/.conda/envs/smithy/bin/conda-smithy", line 10, in <module>
    sys.exit(main())
  File "/home/jaime/.conda/envs/smithy/lib/python3.8/site-packages/conda_smithy/cli.py", line 613, in main
    args.subcommand_func(args)
  File "/home/jaime/.conda/envs/smithy/lib/python3.8/site-packages/conda_smithy/cli.py", line 419, in __call__
    self._call(args, tmpdir)
  File "/home/jaime/.conda/envs/smithy/lib/python3.8/site-packages/conda_smithy/cli.py", line 424, in _call
    configure_feedstock.main(
  File "/home/jaime/.conda/envs/smithy/lib/python3.8/site-packages/conda_smithy/configure_feedstock.py", line 1953, in main
    render_azure(env, config, forge_dir, return_metadata=True)
  File "/home/jaime/.conda/envs/smithy/lib/python3.8/site-packages/conda_smithy/configure_feedstock.py", line 1208, in render_azure
    return _render_ci_provider(
  File "/home/jaime/.conda/envs/smithy/lib/python3.8/site-packages/conda_smithy/configure_feedstock.py", line 611, in _render_ci_provider
    migrated_combined_variant_spec = migrate_combined_spec(
  File "/home/jaime/.conda/envs/smithy/lib/python3.8/site-packages/conda_smithy/configure_feedstock.py", line 563, in migrate_combined_spec
    combined_spec = variant_add(combined_spec, migration)
  File "/home/jaime/.conda/envs/smithy/lib/python3.8/site-packages/conda_smithy/variant_algebra.py", line 238, in variant_add
    return VARIANT_OP[operation](v1, v2)
  File "/home/jaime/.conda/envs/smithy/lib/python3.8/site-packages/conda_smithy/variant_algebra.py", line 140, in op_variant_key_add
    new_keys = variant_key_set_union(
  File "/home/jaime/.conda/envs/smithy/lib/python3.8/site-packages/conda_smithy/variant_algebra.py", line 115, in variant_key_set_union
    return sorted(out_v, key=partial(_version_order, ordering=ordering))
  File "/home/jaime/.conda/envs/smithy/lib/python3.8/site-packages/conda_smithy/variant_algebra.py", line 66, in _version_order
    return ordering.index(v)
ValueError: '3.8' is not in list
```

**Comment by isuruf** [comment]
It's not `cuda110.yaml`. You need `recipe/conda_build_config.yaml` from this PR

**Comment by jaimergp** [comment]
Oops, ok. This should be it now:

https://github.com/jaimergp-feedstocks/ambertools-feedstock/commit/f5c261d658fb0c19eb90bc0fdc304f392cf91d92

**Comment by beckermr** [comment]
Remove the cuda110 file and try again.

**Comment by jaimergp** [comment]
Ok, wow, I should get some sleep 😄 

https://github.com/jaimergp-feedstocks/ambertools-feedstock/commit/486d8808c5a344627c123dc46a8f327e8227bb08

**Comment by beckermr** [comment]
Alright! LGTM!

**Pull request merged**